tests/integration/workflows/go_modules/test_go.py

```diff
@@ -112,6 +112,19 @@ def test_builds_arm64_architecture(self):
             options={"artifact_executable_name": "no-deps-main-arm64"},
             architecture="arm64",
         )
-
         pathname = Path(self.artifacts_dir, "no-deps-main-arm64")
         self.assertEqual(get_executable_arch(pathname), "AArch64")
+
+    def test_builds_with_trimpath(self):
```

Comment:
Can we add a test to show the difference between when trim path is enabled vs not enabled?

Reply:
@sriram-mv  @mndeveci I have added this test case. I wrote a utility that will return the md5 hexdigest of the built binaries. I created a copy of the `no-deps` go project in `testdata`. By building an exact copy of that project with `-trimpath` enabled we can assert that their md5 hex digests are the same. Similarly, we can assert that they will not be equal without the `-trimpath` flag. This was the most straightforward way that I could think of. Hopefully it is sufficient.